lib/component.js

```diff
@@ -6,97 +6,89 @@ const inquirer = require('inquirer');
-          choices: patternDir.map(item => path.basename(item)),
-        },
-        {
-          type: 'input',
-          name: 'component_folder_sub',
-          message: 'Include subfolder or leave blank',
-        },
-      ];
-      return inquirer
-        .prompt(questions)
-        .then(answers =>
-          Object.assign(answers, { component_name: componentName })
-        );
-    })
-    .then(answers => {
-      const componentName = answers.component_name;
-      const componentTitle = answers.component_title;
-      const componentDocumentation = answers.document;
-      const componentLocation = path.join(
-        patternSrc,
-        answers.component_folder,
-        machineName(answers.component_folder_sub),
-        machineName(componentName)
-      );
-      const output = `---
+  const { componentName } = await inquirer.prompt(questions);
+  return componentName.trim();
+}
+
+async function getComponentDetails(componentName, patternDir) {
+  const defaultComponentTitle = humanName(componentName);
+  const detailedQuestions = [
+    {
+      type: 'input',
+      name: 'componentTitle',
+      message: 'What is the human-readable title of your component?',
+      default: defaultComponentTitle,
+    },
+    {
+      type: 'list',
+      name: 'componentFolder',
+      message: 'Component Location',
+      choices: patternDir.map(item => path.basename(item)),
+    },
+    {
+      type: 'input',
+      name: 'componentFolderSub',
+      message: 'Include subfolder or leave blank',
+    },
+  ];
+  return await inquirer.prompt(detailedQuestions);
+}
+
+async function init() {
+  const patternSrc = `${process.cwd()}/source/_patterns/`;
+  const patternDir = await getDirectories(patternSrc);
+  const componentName = await getMachineName();
+  const {
+    componentTitle,
+    componentFolder,
+    componentFolderSub,
+  } = await getComponentDetails(componentName, patternDir);
+  const componentLocation = path.join(
+    patternSrc,
+    componentFolder,
+    machineName(componentFolderSub),
+    machineName(componentName)
+  );
+  const output = `---
 Component Name: ${componentName}
 Component Title: ${componentTitle}
 Component Location: ${componentLocation}
 `;
-      console.log(output);
-
-      const confirm = [
-        {
-          type: 'confirm',
-          name: 'confirm',
-          message: 'Is this what you want?',
-        },
-      ];
-
-      inquirer.prompt(confirm).then(answers => {
-        if (answers.confirm) {
-          createComponent(
-            componentName,
-            componentTitle,
-            componentLocation,
-            componentDocumentation
-          );
-        } else {
-          console.log('Component cancelled');
-        }
-      });
-    });
+  console.log(output);
+  const confirm = [
+    {
+      type: 'confirm',
+      name: 'confirm',
+      message: 'Is this what you want?',
+    },
+  ];
+  const confirmation = await inquirer.prompt(confirm);
```

Comment:
I'd probably rename variables so that you're able to use destructuring:

```js
const prompt = [...];
const { confirm } = await inquirer.prompt(prompt);
```